### PR TITLE
Fix config system secrets

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Fix for handling config changes. Instead of creating a a secret per config type, the data is added under different keys in the secret for the streamlet deployment.

### Why are the changes needed?

There were some complications in the previous solution that caused out of order updates of the resources.

### Does this PR introduce any user-facing change?

no.

### How was this patch tested?

GKE.

TODO: (in separate PR)
- [ ] translate PodsConfig to Flink resource requirements and env
- [ ] translate PodsConfig to Spark resource requirements and env

### CODE REVIEW

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala [hunk 1]
```diff
@@ -102,15 +118,22 @@ object Action {
...
-  def executed(action: Action[ObjectResource]) =
-    s"Executed ${action.name} action for ${action.resource.kind} ${action.resource.metadata.name} in namespace ${action.resource.namespace}"
+  def executed(action: Action[ObjectResource]) = action.executed
+}
 
+abstract class ResourceAction[T <: ObjectResource] extends Action[T] {
+  def resource: T
+  def resourceName = resource.metadata.name
+  def namespace    = resource.metadata.namespace
+  def executing =
```
Comment (maasg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala:133]: I like the idea of adding some state markers. Yet, I find that this approach could be better structured.
The difference in the messages between `executing` and `executed` is very small every time. It also follows a pattern we could exploit:
`<action> for <identifier>` 
Not really a blocker right now.

Comment (RayRoestenburg) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala:133]: yeah this is just refactoring what we had already.. The delete action was a bit weird that it needed the total resource to delete it, so I modified that.

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala [hunk 2]
```diff
@@ -96,20 +103,54 @@ trait Runner[T <: ObjectResource] {
...
+      .fromConfig(ConfigFactory.parseString(str))
+      .recover {
+        case e =>
+          log.error(
+            s"Detected pod configs in secret '${secret.metadata.name}' that contains invalid configuration data, IGNORING configuration.",
+            e
+          )
+          PodsConfig()
+      }
+      .getOrElse(PodsConfig())
```
Comment (maasg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala:134]: Given that `recover` returns a `PodConfig()` isn't `get` sufficient here?

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala [hunk 3]
```diff
@@ -96,20 +103,54 @@ trait Runner[T <: ObjectResource] {
...
+    Try(ConfigFactory.parseString(str))
+      .recover {
+        case e =>
+          log.error(
+            s"Detected runtime config in secret '${secret.metadata.name}' that contains invalid configuration data, IGNORING configuration.",
+            e
+          )
+          ConfigFactory.empty
+      }
+      .getOrElse(ConfigFactory.empty)
```
Comment (maasg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala:148]: same here

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala [hunk 4]
```diff
@@ -188,7 +229,8 @@ object PodsConfig {
    *  }}}
    */
   def fromConfig(config: Config): Try[PodsConfig] =
-    Try(PodsConfig(asConfigObjectToMap[PodConfig](config.getConfig("kubernetes.pods"))))
+    if (config.isEmpty) Try(PodsConfig())
```
Comment (maasg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala:232]: `Success(PodsConfig())` ?

File: integration-tests/swiss-knife/build.sbt [hunk 5]
```diff
@@ -24,6 +24,9 @@ lazy val swissKnife = (project in file("."))
       
     )
 lazy val app = (project in file("./app"))
+  .settings(
+    name:= "swiss-knife"
```
Comment (maasg) [inline integration-tests/swiss-knife/build.sbt:28]: I have this on a branch as well. 3rd time today, I think :-)

**Comment by maasg** [comment]
merged :-)

**Pull request merged**